### PR TITLE
Add conditional recommendations to the Jetpack assistant

### PR DESCRIPTION
**Pull request by jboland88**
#### Changes proposed in this Pull Request:

* This PR adds logic to show recommendations in the Jetpack assistant based on actions that occur on a site.
* In this PR, publishing a new post will enable a recommendation for Publicize to show in the assistant interface.

#### Jetpack product discussion

p1HpG7-dON-p2

#### Does this pull request change what data or activity we track or use?

* Starts keeping track of which recommendations have been viewed as part of the `recommendations_data` option.
* Adds a new `recommendations_conditional` option, which saves the conditional recommendations that are enabled for the site.
* This PR adds two new steps to the recommendations flow (Publicize and VideoPress). These steps do not add any new analytics events, but they use current analytics events to record activity on the related step screens and summary screen, passing the step/feature slug as a property to the event. The existing analytics events used for these new steps are:
```
// Individual Steps
jetpack_recommended_feature_learn_more_click
jetpack_recommended_feature_enable_click
jetpack_recommended_feature_decide_later_click

// Summary screen
jetpack_recommendations_summary_configure_click
jetpack_recommendations_summary_learn_more_click
jetpack_recommendations_summary_enable_click
```

#### Testing instructions:

To reset the assistant state while testing, you can use the following commands ( you can run these with `jetpack docker` if using the Docker development environment )
* `wp jetpack options delete recommendations_data`
* `wp jetpack options delete recommendations_conditional`

If your test site already has the Publicize module enabled, you will not receive the new recommendation as outlined in the test plan below. Be sure to disable the Publicize module before testing.

---

* Checkout this branch on your Jetpack test site.
* Visit the Jetpack dashboard. Note that the "Recommendations" tab in the dashboard navigation does not show any badge indicating that there is a new recommendation.
Screen Shot 2022-03-21 at 2 27 58 PM

* Now, go create and publish a new post (or switch a post to draft and back to publish)
* After the post is published, navigate back to the Jetpack Dashboard. You should see a red badge with a "1" next to the Recommendations tab in the navigation on the Jetpack dashboard - indicating a new recommendation is available.
Screen Shot 2022-03-11 at 4 00 23 PM

* Click on the "Recommendations" tab in the navigation - you should now see a recommendation to enable the Publicize feature. There should be a green "New" badge that shows on this step next to the progress bar.
Screen Shot 2022-03-11 at 4 58 50 PM

* Proceed through the rest of the flow to the summary screen. On the summary screen, you should see the "Publicize" feature listed and a green "New" badge next to it in the list.
Screen Shot 2022-03-11 at 4 59 06 PM

* If you reload the page on the summary screen, the green "New" badge next to Publicize in the summary feature list and the red "1" indicator next to "Recommendations" in the dashboard navigation  should no longer show.
* Test that the buttons to enable Publicize from both the individual recommendation screen and the summary screen work and enable the correct module.
* Automated unit tests should be passing
* Visuals should match this reference: G151IYl8Lm4f0NjgLpR943-fi-2132%3A39570

### CODE REVIEW

File: projects/plugins/jetpack/_inc/client/recommendations/feature-summary/index.jsx
Comment (monsieur-z) [inline projects/plugins/jetpack/_inc/client/recommendations/feature-summary/index.jsx:95]: A comment for translators might be useful here.

**Review by monsieur-z — CHANGES_REQUESTED** [review]
Every steps mentioned in the instructions worked as expected, except from the count and _New_ badges disappearing after refreshing the page. I could still see them in the summary page.

Going through the flow, I'm wondering if the Publicize step should lead straight to the summary (after either skipping it, or enabling the feature), or if we should display the other steps of the wizard. I don't have a strong opinion about either. Do you know what's the rationale behind it?

Code changes look sensible to me.

I noticed 2 small styling issues. The count badge is not exactly vertically aligned, and the icon of the external link is underlined when it shouldn't. I think at least the latter should be addressed.

File: projects/plugins/jetpack/_inc/client/components/navigation/index.jsx
Comment (jeherve) [inline projects/plugins/jetpack/_inc/client/components/navigation/index.jsx:117]: I wonder if it may make more sense to make that a new translatable string with a placeholder, using `createInterpolateElement`. I think in some languages (RTL), they may expect the bubble on the other side of the word.

File: projects/plugins/jetpack/_inc/client/components/section-nav/style.scss
Comment (jeherve) [inline projects/plugins/jetpack/_inc/client/components/section-nav/style.scss:358]: Maybe you'd be able to pull this from https://github.com/Automattic/jetpack/blob/3238c5d6df6727ce64fc18763a0025f10920a317/projects/plugins/jetpack/_inc/client/scss/variables/_colors.scss#L48

File: projects/plugins/jetpack/_inc/client/recommendations/feature-utils.js
Comment (jeherve) [inline projects/plugins/jetpack/_inc/client/recommendations/feature-utils.js:49]: Should we point to Calypso in this case, where folks will be able to connect new services to their site? `getRedirectUrl` should be helpful for this:
https://github.com/Automattic/jetpack/blob/39fa657bfe0aaca179275331752fd99a74ac1c41/projects/plugins/jetpack/_inc/client/sharing/publicize.jsx#L49

File: projects/plugins/jetpack/_inc/client/recommendations/feature-utils.js
Comment (jeherve) [inline projects/plugins/jetpack/_inc/client/recommendations/feature-utils.js:187]: I would recommend using `getRedirectUrl` here as well. PCYsg-pY7-p2 should help you create a new link.

File: projects/plugins/jetpack/_inc/lib/class-jetpack-recommendations.php
Comment (jeherve) [inline projects/plugins/jetpack/_inc/lib/class-jetpack-recommendations.php:120]: Since we're all in the same class you should be able to simplify things here.

File: projects/plugins/jetpack/_inc/client/components/section-nav/style.scss
Comment (jboland88) [reply projects/plugins/jetpack/_inc/client/components/section-nav/style.scss]: Ahh, nice. Thanks for the reference 👍

File: projects/plugins/jetpack/_inc/client/recommendations/feature-utils.js
Comment (jboland88) [reply projects/plugins/jetpack/_inc/client/recommendations/feature-utils.js]: Good idea, much more helpful to point to Calypso here.

File: projects/plugins/jetpack/_inc/lib/class-jetpack-recommendations.php
Comment (jboland88) [reply projects/plugins/jetpack/_inc/lib/class-jetpack-recommendations.php]: I've made some changes here - LMK if you had another thought on how to improve this :)

**Comment by jboland88** [comment]
@monsieur-z - Thanks for the review!

> Every steps mentioned in the instructions worked as expected, except from the count and New badges disappearing after refreshing the page. I could still see them in the summary page.

Hmmm, I was not able to reliably reproduce this. It happened once to me 🤔  But 3 other times it worked correctly. Are you still getting this behavior?

> Going through the flow, I'm wondering if the Publicize step should lead straight to the summary (after either skipping it, or enabling the feature), or if we should display the other steps of the wizard.

Good question, inquiring about this to get some more input: pbNhbs-2b7-p2#comment-7373

> I noticed 2 small styling issues. The count badge is not exactly vertically aligned, and the icon of the external link is underlined when it shouldn't. I think at least the latter should be addressed.

I got the underline on the link sorted, The count badge looks to still to be about 1px taller than the line height of the "Recommendations" text. It looks like other update badges in wp-admin are also top aligned and slightly taller than their text 👍

**Comment by jboland88** [comment]
@monsieur-z - I think I've got the inconsistent behavior of the recommendations badge disappearing sorted now :)

**Review by monsieur-z — DISMISSED** [review]
I can now see the notification disappear consistently. Code changes look sensible.

File: projects/plugins/jetpack/_inc/client/components/navigation/index.jsx
Comment (jeherve) [inline projects/plugins/jetpack/_inc/client/components/navigation/index.jsx:128]: That's a lot of span. :) I assume it wasn't possible to do without the extra span?

**Review by jeherve — DISMISSED** [review]
This is going to need a rebase before we can merge I'm afraid.

**Comment by jboland88** [comment]
@jeherve - thanks for the feedback! I've rebased this one and cleaned up a couple small items.

**Comment by kraftbj** [comment]
@coder-karen @sdixon194 Let's prioritize this for review so we can land it as soon as feasible so it doesn't grow stale during this cycle.

File: projects/plugins/jetpack/_inc/lib/class-jetpack-recommendations.php
Comment (samiff) [inline projects/plugins/jetpack/_inc/lib/class-jetpack-recommendations.php:125]: `get_called_class()` is a new one for me. Just curious, would `self::class` work here the same?

**Review by samiff — DISMISSED** [review]
This tested well for me 👍 

I did notice one design quirk on mobile with the badge:

Screen Shot 2022-04-05 at 5 06 19 PM

But that could always be looked at in a follow-up PR.

**Comment by jboland88** [comment]
Hi @samiff 👋 I had to merge master to resolve some conflicts and it dismissed your approval.

**Review by kraftbj — APPROVED** [review]
Reapproving after catchup

**Review by kraftbj — DISMISSED** [review]

**Review by kraftbj — APPROVED** [review]

**Comment by kraftbj** [comment]
r243251-wpcom

**Pull request merged**
